session/session.go

```diff
@@ -949,22 +952,24 @@ func (s *session) DropPreparedStmt(stmtID uint32) error {
 	return nil
 }
 
-func (s *session) Txn(active bool) kv.Transaction {
+func (s *session) Txn(active bool) (kv.Transaction, error) {
 	if s.txn.pending() && active {
-		// Transaction is lazy intialized.
+		// Transaction is lazy initialized.
 		// PrepareTxnCtx is called to get a tso future, makes s.txn a pending txn,
 		// If Txn() is called later, wait for the future to get a valid txn.
 		txnCap := s.getMembufCap()
 		if err := s.txn.changePendingToValid(txnCap); err != nil {
-			s.txn.fail = errors.Trace(err)
-		} else {
-			s.sessionVars.TxnCtx.StartTS = s.txn.StartTS()
+			log.Error("active transaction fail, err = ", err)
```

Comment:
```suggestion
 			log.Errorf("active transaction fail, err = %+v", err)
```
if we need stack here

Reply:
done